src/api/kv2/requests.rs

```diff
@@ -256,6 +257,7 @@ pub struct SetSecretMetadataRequest {
     pub max_versions: Option<u64>,
     pub cas_required: Option<bool>,
     pub delete_version_after: Option<String>,
+    pub custom_metadata: Option<HashMap<String, String>>,
```

Comment:
Shouldn't this be `Option<HashMap<String, Value>>`? [Official go client](https://github.com/hashicorp/vault/blob/api/v1.9.0/api/kv_v2.go#L25) uses here `map[string]interface{}`

Reply:
Just tried it, it seems only strings are allowed:
```
2023-03-11T20:15:11.460178Z ERROR rustify::endpoint: error=Server returned error
2023-03-11T20:15:11.460201Z ERROR vaultrs::api: Detected errors in API response: [
    "custom_metadata validation failed: '' expected type 'string', got unconvertible type 'map[string]interface {}', value: 'map[foo:bar]'",
]
2023-03-11T20:15:11.460232Z ERROR vaultrs::kv2: error=The Vault server returned an error (status code 400)
```

Reply:
@nicoulaj try `vault kv metadata put -custom-metadata=foo={"bar":"baz"} secret/test`. In go syntax `map[string]interface{}` is dictionary/hashmap with string keys holding any type.